samples/texture_derivative.html

```diff
@@ -0,0 +1,354 @@
+        var indexBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuffer);
+
+        var cubeVertexIndices = [
+          0,  1,  2,      0,  2,  3,    // front
+          4,  5,  6,      4,  6,  7,    // back
+          8,  9,  10,     8,  10, 11,   // top
+          12, 13, 14,     12, 14, 15,   // bottom
+          16, 17, 18,     16, 18, 19,   // right
+          20, 21, 22,     20, 22, 23    // left
+        ];
+
+        // Now send the element array to GL
+
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array(cubeVertexIndices), gl.STATIC_DRAW);
+
+        // -- Init VertexArray
+        var vertexArray = gl.createVertexArray();
+        gl.bindVertexArray(vertexArray);
+
+        var vertexPosLocation = 0; // set with GLSL layout qualifier
+        gl.enableVertexAttribArray(vertexPosLocation);
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexPosBuffer);
+        gl.vertexAttribPointer(vertexPosLocation, 3, gl.FLOAT, false, 0, 0);
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+        var vertexTexLocation = 1; // set with GLSL layout qualifier
+        gl.enableVertexAttribArray(vertexTexLocation);
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexTexBuffer);
+        gl.vertexAttribPointer(vertexTexLocation, 2, gl.FLOAT, false, 0, 0);
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuffer);
+
+        gl.bindVertexArray(null);
+
+        // -- Init Texture
+
+        var imageUrl = '../assets/img/Di-3d.png';
+        var texture;
+        loadImage(imageUrl, function(image) {
+            // -- Init 2D Texture
+            texture = gl.createTexture();
+            gl.activeTexture(gl.TEXTURE0);
+            gl.bindTexture(gl.TEXTURE_2D, texture);
+            gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, false);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+
+            // -- Allocate storage for the texture
+            gl.texStorage2D(gl.TEXTURE_2D, 1, gl.RGB8, 512, 512);
+            gl.texSubImage2D(gl.TEXTURE_2D, 0, 0, 0, gl.RGB, gl.UNSIGNED_BYTE, image);
+
+            requestAnimationFrame(render);
+
+        });
+
+        // -- Initialize render variables
+        var orientation = [0.0, 0.0, 0.0];
+
+        var modelMatrix = mat4.create([
+            0.5, 0.0, 0.0, 0.0,
+            0.0, 0.5, 0.0, 0.0,
+            0.0, 0.0, 0.5, 0.0,
+            0.0, 0.0, 0.0, 1.0
+        ]);
+
+        var viewMatrix = mat4.lookAt(
+            [4, 3, 300],
+            [0, 0, 0],
+            [0, 1, 0]
+            );
+
+        var mvMatrix = mat4.multiply(viewMatrix, modelMatrix);
+        var perspectiveMatrix = mat4.perspective(0.45, 1, 0.1, 1000.0);
```

Comment:
Try a tigther far-to-near ratio.e.g., a near plane of `1.0` instead of `0.1`.

For more than you'll ever want to know about z-fighting, see http://cesiumjs.org/massiveworlds/index.html#UsingMultipleFrustumsforMassiveWorlds

@shrekshao @trungtle I would use this as a default near/far for all examples.